lib/rspec/mocks/any_instance/proxy.rb

```diff
@@ -91,12 +91,12 @@ def perform_proxying(method_name, args, block, &target_proxy_block)
       # `targets` will typically contain 1 of the `AnyInstance::Recorder`
       # return values and N `MessageExpectation` instances (one per instance
       # of the `any_instance` klass).
-      class FluentInterfaceProxy
+      class FluentInterfaceProxy < BasicObject
         def initialize(targets)
           @targets = targets
         end
 
-        if RUBY_VERSION.to_f > 1.8
+        if ::RUBY_VERSION.to_f > 1.8
```

Comment:
I'm on the fence about this, its not ambiguous so it just creates churn

Reply:
If we have subclassed from BasicObject, then we do not have access to any constants (as root level constants are defined on object). 

This is not a style change. Without this change I get error:

```
NameError:
  uninitialized constant RSpec::Mocks::AnyInstance::FluentInterfaceProxy::RUBY_VERSION
```